test/sv/interfaces.mlir

```diff
@@ -35,6 +35,16 @@ func @test() {
   // CHECK-NEXT:   sv.interface.modport @dataflow_out ("output" @data, "output" @valid, "input" @ready)
   // CHECK-NEXT: }
 
-  // CHECK-NEXT: return
-  return
+  rtl.externmodule @Rcvr (%m: !sv.interface.modport<@handshake_example::@dataflow_in>)
+  // CHECK-LABEL: rtl.externmodule @Rcvr(!sv.interface.modport<@handshake_example::@dataflow_in> {rtl.name = "m"})
+
+  rtl.module @Top () {
+    %iface = rtl.wire : !sv.interface<@handshake_example>
```

Comment:
Would rtl.instance be a more appropriate style handler?  wire is kind of flat, instance expects things which have (mixed) direction.
then 
%i = sv.interface.instance: !sv.interface<@handshake_example>
and to pass around modports:
%j = sv.interface.port %i, @dataflow_in : (!sv.interface<@handshake_example>) -> !sv.interface.modport<@dataflow_in>
too hook up to a modport:
%foo = sv.portconnect %j (%data_v, %valid_v) : (i32, i1) -> i1

Soo portconnect would produce rtl usable things, and the modport type would be what was passed though modules.

Reply:
I was debating this. I'm pretty sure "wire" is a misnomer and it actually means "declare this type". If it were supposed to mean wire literally, it would only accept int args. OTOH, it's part of the RTL dialect, which does have traditional SSA semantics whereas the SV dialect is tending towards representing SystemVerilog AST in a "structural" non-SSA semantics form.

Come to think of it, modport semantics don't mix well with RTL dialect semantics (which requires that module arguments be either input or output). Soo... crap. In order to model this properly (to not abuse `rtl.[extern]module`), we need a `sv.[extern]module` and corresponding `sv.instance` ops. @mikeurbach / @darthscsi / @lattner Am I right about this?

Reply:
Sorry, I must sound like an idiot. My goal was to be able to output SystemVerilog which contains a module instantiation with a modport port. I was realizing that this isn't possible with the RTL dialect so I'd have to do something like what @darthscsi recommended.

Reply:
In a sense modports are always inputs in SV, even if they have outputs in them.  I haven't thought about the implications of an input which has something accessible through it which is an output.

Reply:
I see what you're saying but in another (more important sense IMO) they're not since you'll need to assign (or set) the outputs of that modport.

>  I haven't thought about the implications of an input which has something accessible through it which is an output.

The way I deal with backpressure on the receiver side in ESI (since I want to model it as an input in the RTL dialect) is that it must be [unwrapped and that operation](https://github.com/llvm/circt/blob/3eef7df131ca2ab5933c56509004f26f1d4cee02/test/esi/connectivity.mlir#L13) exposes the backpressure as an argument and the valid and data are results. Something like this _could_ work for modports though in the general case, I don't think it's appropriate but I don't feel strongly about it.

Reply:
I'm thinking about this too... I didn't think this PR was too far off using the RTL dialect. I'm going to check out this branch and play around with some of the ideas I've had in review unless you're already going in a different direction @teqdruid .

Reply:
I'm still vacillating on it, but haven't made any changes so feel free.